lib/widgets/breez_avatar_dialog.dart

```diff
@@ -236,12 +219,23 @@ Widget breezAvatarDialog(BuildContext context, UserProfileBloc userBloc) {
   );
 }
 
-// Do not pop dialog if there's a upload in progress
-Future<bool> _onWillPop(UPLOAD_STATE _state) async {
-  if (_state == UPLOAD_STATE.UPLOAD_IN_PROGRESS) {
-    return false;
+Future<bool> _uploadImage(File _pickedImage, UserProfileBloc userBloc) async {
```

Comment:
I think it is better not to return bool but rather just return error from the future in case the upload failed.
It will simplify the function:

Reply:
[Done](https://github.com/breez/breezmobile/pull/283/commits/44aefdb0abd3388ff3b6b1b9b7bd6b220941602b)